include/aikido/perception/PerceptionModule.hpp

```diff
@@ -31,9 +33,18 @@ class PerceptionModule
   /// otherwise.
   virtual bool detectObjects(
       const aikido::planner::WorldPtr& env,
+      std::vector<DetectedObject>* detectedObjects,
       ros::Duration timeout,
       ros::Time timestamp)
       = 0;
+
+  bool detectObjects(
```

Comment:
Why are these two near-duplicates necessary? If this is an optional out-param, can we just have `const std::vector<DetectedObject>& detectedObject = std::vector<DetectedObject>()`?

Reply:
Doesn't work, then we can't add any new elements.